### PR TITLE
Devendor `frozendict`

### LINKED ISSUE #13767: Devendor `frozendict`
### What is the idea?

Mark `conda._vendor.frozendict` as pending deprecation.

### Why is this needed?

All other vendored libraries have already been devendored. This is the last one.

```[tasklist]
- [x] Package `frozendict` for `defaults`
- [ ] https://github.com/conda/conda/pull/13766
```

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

`frozendict` is the last vendored dependency not marked for deprecation.

Resolves #13767 

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Review by jezdez — APPROVED** [review]

File: conda/resolve.py [hunk 1]
```diff
@@ -43,6 +42,11 @@
 from .models.records import PackageRecord
 from .models.version import VersionOrder
 
+try:
+    from frozendict import frozendict
+except ImportError:
+    from ._vendor.frozendict import FrozenOrderedDict as frozendict
```
Comment (dholth) [inline conda/resolve.py:48]: Should these be excluded from coverage tests?

**Review by dholth — COMMENTED** [review]
Thanks, great changes.

**Comment by jezdez** [comment]
Oh dang, this broke the canary builds :(

**Comment by kenodegard** [comment]
@jezdez yeah that was expected, `frozendict` isn't on `main` yet

**Comment by jezdez** [comment]
> @jezdez yeah that was expected, `frozendict` isn't on `main` yet

Makes sense, you even said it on Slack, distance my brain was able to travel: 0

**Pull request merged**